go-controller/pkg/network-controller-manager/network_controller_manager.go

```diff
@@ -48,13 +49,15 @@ type NetworkControllerManager struct {
 	// Supports OVN Template Load Balancers?
 	svcTemplateSupport bool
 
-	stopChan chan struct{}
-	wg       *sync.WaitGroup
-
+	stopChan                 chan struct{}
+	wg                       *sync.WaitGroup
+	portCache                *ovn.PortCache
```

Comment:
its a shared cache now where as before it was not. I needed to do this so that EIP controller can lookup ports from all networks. I was a little concerned with this because of lock contention but i think it should be ok.

The shared port cache is at this level because NCM controls the creation of new networks and needs to know where the cache is located in-order to deliver this info to the networks it creates and therefore eip controller has access to all the networks ports.

Reply:
@npinaeva can you ack youve seen this?

Reply:
yep, that should be fairly simple to make it lockable per-pod-name, @martinkennelly ping me if you need help/reviews

Reply:
Note: create an issue to track this and self assign.